test/tests/ha/rackdh_ha_resource_test.py

```diff
@@ -0,0 +1,153 @@
+            self.assertTrue(self.is_service_up("pacemaker", vmnum), "Pacemaker not running on node {}".format(vmnum))
+
+    def test011_check_default_configuration(self):
+        err = ''
+        for vmnum in range(1, self.number_of_vms + 1):
+            # Check for default configuration
+            if self.has_online_nodes(vmnum, 'ora'):
+                err += (str(vmnum) + ', ')
+        self.assertEqual(len(err), 0, "The following VMs have the default configuration: {}"
+                         .format(err.rstrip(", ")))
+
+    def test012_check_node_status(self):
+        err = ''
+        for vmnum in range(1, self.number_of_vms + 1):
+            for node_number in range(1, self.number_of_vms + 1):
+                # Determine if this node is online
+                if not self.has_online_nodes(vmnum, 'node{}'.format(node_number)):
+                    err += 'node{} on vm{}, '.format(node_number, vmnum)
+        self.assertEqual(len(err), 0, "The following nodes are not online: {}"
+                         .format(err.rstrip(", ")))
+
+    def test020_check_and_collect_online_nodes(self):
+        # determine node status and collect active nodes in cluster
+        err = ''
+        for vmnum in range(1, self.number_of_vms + 1):
+            command = "crm_mon -X | grep 'node{}.*online=.true' -q".format(vmnum)
+            status = fit_common.remote_shell(command, vmnum=vmnum)['exitcode']
+            if status == 0:
+                active_node_list.append(vmnum)
+            else:
+                err += 'node{}, '.format(vmnum)
+        if FAIL_ON_NODE_COLLECTION:
+            self.assertEqual(len(err), 0, 'The following nodes are offline: {}'
+                             .format(err.rstrip(", ")))
+
+    @depends(after='test020_check_and_collect_online_nodes')
+    def test030_check_mongo_resource(self):
+        err = ''
+        for mongo_number in range(1, self.number_of_services + 1):
+            mongo_name = 'docker_mongo_{}'.format(mongo_number)
+            if not self.has_resource(active_node_list[0], mongo_name):
+                err += (mongo_name + ', ')
+        self.assertEqual(len(err), 0, "The following Mongo resources are not present: {}"
+                         .format(err.rstrip(", ")))
+
+    @depends(after='test020_check_and_collect_online_nodes')
+    def test031_check_mongo_colocation(self):
+        err = ''
+        for mongo_number in range(1, self.number_of_services + 1):
+            command = "crm config show | grep 'mongo{0}.*inf:.*mongo_{0}.*addr_{0}'" \
+                      .format(mongo_number)
+            status = fit_common.remote_shell(command, vmnum=active_node_list[0])['exitcode']
+            if status != 0:
+                err += 'docker_mongo_{0} on mongo_addr_{0}, '.format(mongo_number)
+        self.assertEqual(len(err), 0, 'The following Mongo coLocations are missing: {}'
+                         .format(err.rstrip(", ")))

```

Comment:
I think we also have to check resource "mongo_addr_" before checking colocation between 'docker_mongo_' and 'mongo_addr_'